Assets/2_MainScene/S2_Scripts/S2_EnemyScripts/S2_EnemyManager.cs

```diff
@@ -0,0 +1,360 @@
+using System.Collections;
```

Comment:
Script is taking care of Enemy Spawning, Behavior, and Managing: Should be exploded into Spawner, EnemyMovement, and Manager

Reply:
Use of Instances would be required, the exploded scripts would hold most of the required information and there is not a big hit (if any) with calling Script.Instance

For this specific project that is how it is set-up